scripts/deletePost.js

```diff
@@ -0,0 +1,23 @@
+import { POSTS_API_URL } from "./constants.js";
+import { fetchData } from "./fetchData.js";
+import { singlePostId } from "./utils/getSinglePost.js";
+
+const deleteButton = document.querySelector('#delete-button');
+
+
+async function deletePost() {
+    const postId = singlePostId();
+    const response = await fetchData(`${POSTS_API_URL}/${postId}`, {
+        method: 'DELETE',
+        headers: {
+            'Content-type': 'application/json'
+        },
+    }, true);
+    console.log(response);
+
+}
+
+deleteButton.addEventListener('click', async (event) => {
+    event.preventDefault();
+    await deletePost();
```

Comment:
I would probably redirect to the list of posts page here, otherwise the user won't know the delete was a success as nothing the in UI changes.